nequip/train/metrics.py

```diff
@@ -53,34 +56,40 @@ def __init__(
     ):
 
         self.running_stats = {}
-        self.per_species = {}
+        self.params = {}
         self.funcs = {}
         self.kwargs = {}
         for component in components:
 
             key, reduction, params = Metrics.parse(component)
 
-            functional = params.pop("functional", "L1Loss")
+            params["PerSpecies"] = params.get("PerSpecies", False)
+            params["PerAtom"] = params.get("PerAtom", False)
+
+            param_hash = Metrics.hash_component(component)
```

Comment:
why is this necessary instead of the reduction name?

Or just making it a list and indexing it in order?

Reply:
it turns out that I want to have the same reduction, with and without the "per atom" flag. And in case we have more different things in the future, it's easier to just hash the whole dict.